src/main/java/com/faforever/client/login/LoginController.java

```diff
@@ -396,6 +302,49 @@ public void onDownloadUpdateButtonClicked() {
         });
   }
 
+  public void onLoginButtonClicked() {
+    initializeFuture.join();
+
+    clientProperties.getServer()
+        .setHost(serverHostField.getText())
+        .setPort(Integer.parseInt(serverPortField.getText()));
+
+    clientProperties.getReplay()
+        .setRemoteHost(replayServerHostField.getText())
+        .setRemotePort(Integer.parseInt(replayServerPortField.getText()));
+
+    clientProperties.getIrc()
+        .setHost(ircServerHostField.getText())
+        .setPort(Integer.parseInt(ircServerPortField.getText()));
+
+    clientProperties.getApi().setBaseUrl(apiBaseUrlField.getText());
+    clientProperties.getOauth().setBaseUrl(oauthBaseUrlField.getText());
+
+    OAuthValuesReceiver oAuthValuesReceiver = applicationContext.getBean(OAuthValuesReceiver.class);
+    oAuthValuesReceiver.getRedirectUri().thenAccept(redirectUri -> {
+      clientProperties.getOauth().setRedirectUri(redirectUri);
+      platformService.showDocument(userService.getHydraUrl());
+    });
+    oAuthValuesReceiver.receiveValues().thenAccept(values -> {
+      String actualState = values.getState();
+      String expectedState = userService.getState();
+      if (!expectedState.equals(actualState)) {
+        handleInvalidSate(actualState, expectedState);
+        return;
+      }
+      loginWithCode(values.getCode());
+    });
```

Comment:
Exception silently swallowed. I imagine port being blocked could happen and would be good if user got a message instead of nothing happening

Reply:
.expectionally()

Reply:
Fixed